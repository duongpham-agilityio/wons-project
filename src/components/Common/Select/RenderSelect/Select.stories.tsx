import { Meta, StoryObj } from '@storybook/react';
import { dateDicker, productSelector } from '@/mocks';
import RenderSelect from '.';

export default {
  title: 'Components/RenderSelect',
  component: RenderSelect,
} as Meta;

type Story = StoryObj<typeof RenderSelect>;

// Date dicker
export const DateDicker: Story = {
  args: {
    defaultSelectedKey: 'Weekly',
    items: dateDicker,
    className: 'w-[120px]',
  },
};

// Product selector
export const ProductSelector: Story = {
  args: {
    defaultSelectedKey: 'All Products',
    items: productSelector,
    className: 'w-[152px]',
  },
};
