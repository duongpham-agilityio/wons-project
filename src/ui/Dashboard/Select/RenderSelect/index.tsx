'use client';
import { SelectItem } from '@nextui-org/react';
import SelectCustom from '../SelectCustom';

interface IRenderSelect {
  defaultSelectedKey: string;
  items: Array<{ key: string; label: string }>;
  className: string;
  ariaLabel: string;
}

const RenderSelect = ({
  defaultSelectedKey,
  items,
  className,
  ariaLabel,
}: IRenderSelect) => (
  <SelectCustom
    aria-hidden={true}
    aria-label={ariaLabel}
    defaultSelectedKeys={[defaultSelectedKey]}
    color="default"
    size="default"
    className={`ml-5 ${className}`}
  >
    {items.map((item) => (
      <SelectItem key={item.key}>{item.label}</SelectItem>
    ))}
  </SelectCustom>
);

export default RenderSelect;
