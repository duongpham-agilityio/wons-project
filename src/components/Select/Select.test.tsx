import { render } from '@testing-library/react';
import RenderSelect from './renderSelect';

describe('Select Component', () => {
  const defaultProps = {
    defaultSelectedKey: 'key1',
    items: [
      { key: 'key1', label: 'Item 1' },
      { key: 'key2', label: 'Item 2' },
    ],
    className: 'custom-class',
  };
  test('render select', () => {
    const { container } = render(<RenderSelect {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });
});
