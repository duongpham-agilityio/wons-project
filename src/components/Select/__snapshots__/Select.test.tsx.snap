// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Component render select 1`] = `
<div>
  <div
    class="group flex flex-col w-full transition-background motion-reduce:transition-none !duration-150 group relative justify-end ml-5 custom-class"
    data-filled="true"
    data-has-value="true"
    data-slot="base"
  >
    <div
      aria-hidden="true"
      data-a11y-ignore="aria-hidden-focus"
      data-testid="hidden-select-container"
      style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
    >
      <input
        style="font-size: 16px;"
        tabindex="-1"
        type="text"
        value="key1"
      />
      <label>
        <select
          size="2"
          tabindex="-1"
          title=""
        >
          <option />
          <option
            value="key1"
          >
            Item 1
          </option>
          <option
            value="key2"
          >
            Item 2
          </option>
        </select>
      </label>
    </div>
    <div
      class="w-full flex flex-col"
      data-slot="mainWrapper"
    >
      <button
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="react-aria-:r7: "
        class="relative px-3 gap-3 w-full inline-flex flex-row items-center shadow-sm tap-highlight-transparent group-data-[focus=true]:bg-default-100 min-h-10 rounded-medium data-[hover=true]:bg-default-50 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 bg-white border-1 border-gray-50 py-3.5 h-auto"
        data-slot="trigger"
        id="react-aria-:r2:"
        type="button"
      >
        <div
          class="inline-flex h-full w-[calc(100%_-_theme(spacing.6))] min-h-4 items-center gap-1.5 box-border"
          data-slot="innerWrapper"
        >
          <span
            class="w-full text-left rtl:text-right text-small truncate group-data-[has-value=true]:text-default-foreground font-bold text-black"
            data-slot="value"
            id="react-aria-:r7:"
          >
            Item 1
          </span>
        </div>
        <svg
          aria-hidden="true"
          class="absolute end-3 w-4 h-4 transition-transform duration-150 ease motion-reduce:transition-none data-[open=true]:rotate-180"
          data-slot="selectorIcon"
          fill="none"
          focusable="false"
          height="1em"
          role="presentation"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="m6 9 6 6 6-6"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
